Add Azure Red Hat OpenShift (ARO) distro and doc set

adellape opened the pull request:
Taking over from https://github.com/openshift/openshift-docs/pull/14438.

- Changed instances of arho/ARHO to aro/ARO.
- Removed unrelated SSSD topic changes.

Internal preview: http://file.rdu.redhat.com/~adellape/050119/arho/welcome/index.html

mgahagan73 commented:
I've checked the internal preview, here are the issues I've found:
Legal page says copyright 2017 (should it be 2019?)

getting_started/developers_console.html and getting_started/developers_cli.html
language/tutorials table has no implementation example for Java. I know we removed the Wildfly images but didn't know if we could replace it with the Jboss offering or something similar. 

architecture/infrastructure_components/web_console.html and architecture/additional_concepts/other_api_objects.html are blank pages

Version numbers:
architecture/infrastructure_components/kubernetes_infrastructure.html
doc says: Azure Red Hat OpenShift uses Kubernetes 1.10 and Docker 1.13.
kube version on cluster: v1.11.0
docker version on cluster: 1.13.1

jim-minter commented:
best to say kubernetes v1.11, not v1.11.0.

adellape commented:
Thanks @mgahagan73 @jim-minter! Preview build updated: http://file.rdu.redhat.com/~adellape/050119/arho/welcome/index.html

> Legal page says copyright 2017 (should it be 2019?)

Fixed via https://github.com/openshift/openshift-docs/pull/14686/.

> getting_started/developers_console.html and getting_started/developers_cli.html
> language/tutorials table has no implementation example for Java. I know we removed the Wildfly images but didn't know if we could replace it with the Jboss offering or something similar.

Will follow-up on this separately.
 
> architecture/infrastructure_components/web_console.html and architecture/additional_concepts/other_api_objects.html are blank pages

Fixed the missing `openshift-aro` ifdefs.

> Version numbers:
> architecture/infrastructure_components/kubernetes_infrastructure.html
> doc says: Azure Red Hat OpenShift uses Kubernetes 1.10 and Docker 1.13.
> kube version on cluster: v1.11.0
> docker version on cluster: 1.13.1

Updated to `1.11` and `1.13.1`.

adellape commented:
/cherrypick enterprise-3.11

vikram-redhat commented:
@adellape this shouldn't have merged the distro_map changes. It also undid the changes for the new dedicated 4.1 branch in the distro map.

I have undone this via: https://github.com/openshift/openshift-docs/pull/14715
